docs/admin/authorization.md

```diff
@@ -285,6 +285,27 @@ restricting access to the REST api.
 For further documentation refer to the authorization.v1beta1 API objects and
 plugin/pkg/auth/authorizer/webhook/webhook.go.
 
+## Keystone Mode
+
+When specified, mode `Keystone` causes Kubernetes to validate users
+authenticated with the experimental Keystone Token Authentication plugin
+against the the Keystone Role and Keystone Project ID specified by the
+following api-server command line flags:
+ * `--experimental-authorization-keystone-single-projectid="<The Keystone Project ID>"`
+ * `--experimental-authorization-keystone-role="<The keystone Role Name, such as _member_>"`
+
+This allows a single specified Keystone Project access to the entire set of
+namespaces in the Kubernetes cluster. AKA: Single Tenant mode. This is
+intended to allow any Project to create their own dedicated Kubernetes cluster
+and enable access to it using their existing Keystone credentials.
+
+Restricting Keystone Projects to namespaces, AKA: Multi Tenant Mode, is not
```

Comment:
Hmm.. I can send another patch to put our code into upstream once yours get
in..

On Fri, May 20, 2016 at 11:55 PM, kfox1111 <notifications@github.com> wrote:

> In docs/admin/authorization.md
> <https://github.com/kubernetes/kubernetes.github.io/pull/544#discussion_r64064394>
> :
>
> > @@ -285,6 +285,27 @@ restricting access to the REST api.
> >  For further documentation refer to the authorization.v1beta1 API objects and
> >  plugin/pkg/auth/authorizer/webhook/webhook.go.
> >
> > +## Keystone Mode
> > +
> > +When specified, mode `Keystone` causes Kubernetes to validate users
> > +authenticated with the experimental Keystone Token Authentication plugin
> > +against the the Keystone Role and Keystone Project ID specified by the
> > +following api-server command line flags:
> > + * `--experimental-authorization-keystone-single-projectid="<The Keystone Project ID>"`
> > + * `--experimental-authorization-keystone-role="<The keystone Role Name, such as _member_>"`
> > +
> > +This allows a single specified Keystone Project access to the entire set of
> > +namespaces in the Kubernetes cluster. AKA: Single Tenant mode. This is
> > +intended to allow any Project to create their own dedicated Kubernetes cluster
> > +and enable access to it using their existing Keystone credentials.
> > +
> > +Restricting Keystone Projects to namespaces, AKA: Multi Tenant Mode, is not
>
> @zhouhaibing089 <https://github.com/zhouhaibing089> Fascinating. Any
> chance you can share that code? If not, I can take a stab at implementing
> it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes.github.io/pull/544/files/739f4b8421465a3d8061a0fd9d90cef05011a887#r64064394>
>